Source/missiles.cpp

```diff
@@ -2585,35 +2590,39 @@ void AddStone(int mi, Point /*src*/, Point dst, int /*midir*/, int8_t /*mienemy*
 	int tx;
 	int ty;
 
-	bool faded = false;
+	bool found = false;
 	Missiles[mi]._misource = id;
 	for (int i = 0; i < 6; i++) {
 		int k = CrawlNum[i];
 		int ck = k + 2;
 		for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
 			tx = dst.x + CrawlTable[ck - 1];
 			ty = dst.y + CrawlTable[ck];
-			if (InDungeonBounds({ tx, ty })) {
-				int mid = dMonster[tx][ty];
-				if (mid == 0)
-					continue;
-				mid = mid > 0 ? mid - 1 : -(mid + 1);
-				auto &monster = Monsters[mid];
-				if (mid > MAX_PLRS - 1 && monster._mAi != AI_DIABLO && monster.MType->mtype != MT_NAKRUL) {
-					if (monster._mmode != MM_FADEIN && monster._mmode != MM_FADEOUT && monster._mmode != MM_CHARGE) {
-						faded = true;
-						i = 6;
-						Missiles[mi]._miVar1 = monster._mmode;
-						Missiles[mi]._miVar2 = mid;
-						monster.Petrify();
-						break;
-					}
-				}
-			}
+			if (!InDungeonBounds({ tx, ty }))
+				continue;
+
+			int mid = dMonster[tx][ty];
+			if (mid == 0)
+				continue;
+
+			auto &monster = Monsters[mid > 0 ? mid - 1 : -(mid + 1)];

```

Comment:
```suggestion
			auto &monster = Monsters[abs(mid) - 1];
```
IMO the ternary conditional doesn't make the code easy to understand when it's inlined.